### ISSUE TITLE
BuddyBoss REST API

### ISSUE THREAD

**Issue by eisenwasser** [buddyboss/buddyboss-platform#568]
**Is your feature request related to a problem? Please describe.**
We need to support REST API for accessing all of the important BuddyBoss features and options via external web and mobile apps.

**Describe the solution you'd like**
We are building this already, in another repo as a standalone plugin right now. Once V1 of the APIs is completed we will merge it into the core BuddyBoss Platform and open source it, and we will publish the APIs publicly on our website. We are building it as a plugin so we can easily test the APIs against various branches of BuddyBoss Platform at any moment during development.
_Estimated timeline : 2-3 months_

Private Repo:
https://github.com/buddyboss/buddyboss-platform-api

**Comment by rohanbam** [comment]
Hi, I'm consodering using buddyboss over buddypress for a custom platform we're building for which we require headless CMS capabilities. So I was wondering if there any updates on this and whether the REST API will be made available soon (within a month approximately). Thank you for you work.

**Comment by eisenwasser** [comment]
@rohanbam 

Our API will be made available by the end of May most likely. The absolute latest would be in June but most likely it will be in May.

**Comment by MGParisi** [comment]
Nice to see its released, can we get a link for documentation as soon as it becomes available?

**Comment by chetansatasiya** [comment]
> Nice to see its released, can we get a link for documentation as soon as it becomes available?

@MGParisi You can find here https://www.buddyboss.com/resources/api/

**Comment by gaurav7019** [comment]
How can I authenticate myself as admin to start using the platform API?

**Comment by ThomasCanCode** [comment]
For most API calls, it states in the documentation Permission: LoggedInUser
How can I be the logged in user if I am doing the calls from a php file on the server?

**Comment by Sebastian-Schuchmann** [comment]
Any updates on this. How can I login via the API? There is only an API for Signup, not for logging in?

**Comment by eisenwasser** [comment]
@Sebastian-Schuchmann 

This is the endpoint for login: `buddyboss-app/auth/v1/jwt/token`
That is a POST request and logout would be DELETE to the same endpoint.

**Comment by eisenwasser** [comment]
It's on this resource URL - https://www.buddyboss.com/resources/api/app/#api-Authentication-RequestToken

Maybe we need to improve the description to make it more obvious.

**Comment by Sebastian-Schuchmann** [comment]
When using the App API's I always get a 404: No route was found matching the URL and request method. Is this part of the API only available if you have bought the Buddyboss App? We only have the free version of the Wordpress site.

**Comment by eisenwasser** [comment]
@Sebastian-Schuchmann In our API docs, there are two tabs at the top for "Platform" and "App".

When viewing the Platform APIs, these require BuddyBoss Platform plugin.
When viewing the App APIs, these require BuddyBoss App plugin (paid).

If you are trying to authenticate login just on web, then you don't need our APIs. You can do this through WordPress own REST API methods. We don't have our own special web login API. Our Platform APIs are just for features coming from BuddyBoss Platform. When using BuddyBoss Platform and logging in, we just redirect you to the WordPress login.

**Comment by BoweFrankema** [comment]
Hey Michael @eisenwasser,

I'm working on some REST API stuff for one of my client and I'm still confused on how to **authenticate as a logged in user when not using the BuddyBoss app**. 

What should be the process? There is one other issue here where the following is written:

_There is a jwt token you need to generate and that token is used in API for login so there is an API for that token below.
SiteURL/wp-json/buddyboss-app/auth/v1/jwt/token
above API you need to pass the username, password and app_id then you receive one token. http://i.imgur.com/gtHwmjo.png
once you get the token then you need to pass that token as an accessToken in the header for API. http://i.imgur.com/hkeFgi7.png_

You say this is not needed, you just need to use Basic Authentication (akin to the WP Rest API) and it should work. When I using basic auth I can use the WP JSON v2 API fully to create posts etc etc. But when I try to hit a BuddyBoss Platform endpoint like

```
install + "/wp-json/buddyboss/v1/activity"
```

I get 

```
{
  "code": "bp_rest_authorization_required",
  "message": "Sorry, you are not allowed to create activities.",
  "data": {
    "status": 403
  }
}
```

I've tried various things over the last hours, but I am not getting anywhere.. It would be awesome if your documentation could provide a simple outline on how to get authenticated/logged in for the BuddyBoss Platform (Web version). The rest of the documentation seems excellent :-)

**Comment by BoweFrankema** [comment]
I've had some sleep and review the code of the Rest API and realised the only check was the is_user_logged_in function. So it had to be a problem in the methods I used via the rest API. So I followed the following tutorial which still works with getting the BuddyBoss REST API to work. 

https://apppresser.com/buddypress-rest-api/

I do think it would make sense to add the initial setup steps to your documentation because I believe it would help others!
